content/technical/open-source-as-a-service.md

```diff
@@ -0,0 +1,155 @@
+Title: Your Service is not Open Source
+Date: 2018-02-06 18:35
+Tags: open source, technical
+Slug: open-source-as-a-service
+Category: Technical
+Summary: Sharing the source code to your SaaS is insufficient. We effectively enable users to contribute to the a running service for it to be Open Source.
+
+Sharing the source code to your SaaS is insufficient to make it
+actually be Open Source. Sounds self-contradictory?
+
+Most services that espouse "Open Source", do so by simply
+[throwing the code over the wall](https://github.com/travis-ci/travis-ci).
+It's better than nothing, but really misses the point that powers
+Open Source: enabling users to make a change to the software
+they're using.
+
+Some other Open Source services, such as
+[GitLab.com](https://about.gitlab.com/community/contribute/development/)
+or [ElasticSearch](https://www.elastic.co/about/free-and-open) do include
+the tools used to operate/deploy their service. Pause for applause 👏👏👏
+
+But thats is also insufficient to actually enable Open Source.
+
+In order for a user of the service to begin contributing, they must
+first deploy and operate their own clone: Something that they explicitly
+chose not to do when they elected to use the software as a service.
+
+Even if the user becomes an operator, and deploys a such clone, it
+is devoid of their own data, devoid of other users, of connections
+to other services, of infrastructure scalability, pipelines, monitoring,
+backups, ... you name it.
```

Comment:
This is a bit harsh IMHO. First, the whole network of users, credentials, data etc. are the actual and precious assets of the "primary" SaaS, and exposing/exporting all of that is not only undesirable and inappropriate, it's outright illegal (data protection, private projects, exposing secrets, etc.). Also the whole point of hosting your own instance is that you can work with your own data, test it, and break it.

The second half (scalability, pipelines, monitoring, backups) just means that the deployment scripts are incomplete -- things like metrics collection or database replication should be part of the deployment; of course the "other" side of that (pagers, email etc.) is highly specific to a particular user/company. Also, for test instances you most likely don't want that extra overhead anyway.

I understand that this is primarily an argument for *not* hosting your own instance in this context, and for that it's valid -- but that's already handled in the first paragraph.

Reply:
Good points. I've tried to rewrite this and stick to objective claims. WDYT? Do you feel these paragraphs are more coherent now? I'd be happy for further critique, in areas where my subjectivity remains.